plugins/gtav/structs.h

```diff
@@ -0,0 +1,227 @@
+// Copyright 2022 The Mumble Developers. All rights reserved.
+// Use of this source code is governed by a BSD-style license
+// that can be found in the LICENSE file at the root of the
+// Mumble source tree or at <https://www.mumble.info/LICENSE>.
+
+#ifndef GTAV_STRUCTS
+#define GTAV_STRUCTS
+
+#include <cstdint>
+
+// Reference: https://github.com/Yimura/GTAV-Classes
+
+typedef uint64_t ptr_t;
+
+enum class GameState : int32_t {
+	Invalid = -1,
+	Playing,
+	Died,
+	Arrested,
+	FailedMission,
+	LeftGame,
+	Respawn,
+	InMPCutscene
+};
+
+enum class ModelType : uint8_t { Unk0, Object, Unk1, Unk2, Weapon, Vehicle, Ped, Plant = 129 };
+
+namespace rage {
+template< typename T > union vector3 {
+	T data[3];
+	struct {
+		T x, y, z;
+	};
+};
+
+typedef vector3< float > fvector3;
+
+struct netAddress {
+	uint8_t field4;
+	uint8_t field3;
+	uint8_t field2;
+	uint8_t field1;
+};
+
+struct netPlayerData {
+	uint8_t pad1[8];
+	uint64_t rockstarID;
+	uint8_t pad2[52];
+	netAddress relayIP;
+	uint16_t relayPort;
+	uint8_t pad3[2];
+	netAddress externalIP;
+	uint16_t externalPort;
+	uint8_t pad4[2];
+	netAddress internalIP;
+	uint16_t internalPort;
+	uint8_t pad5[6];
+	uint64_t hostToken;
+	uint64_t peerID;
+	uint64_t rockstarID2;
+	uint8_t pad6[12];
+	char name[20];
+};
+
+#pragma pack(push, 4)
+struct fwEntity {
+	uint8_t pad1[32];
+	ptr_t modelInfo; // CBaseModelInfo *
+	uint8_t pad2[1];
+	int8_t entityType;
+	uint8_t pad3[2];
+	uint8_t invisible;
+	uint8_t pad4[3];
+	ptr_t navigation; // CNavigation *
+	uint8_t pad5[16];
+	ptr_t drawData; // rage::fwDrawData *
+	uint8_t pad6[16];
+	rage::fvector3 right;
+	uint8_t pad7[4];
+	rage::fvector3 forward;
+	uint8_t pad8[4];
+	rage::fvector3 up;
+	uint8_t pad9[4];
+	rage::fvector3 position;
+	uint8_t pad10[52];
+	ptr_t netObject; // rage::netObject *
+	uint8_t pad11[176];
+	uint32_t damageBits;
+};
+#pragma pack(pop)
+} // namespace rage
+
+#pragma pack(push, 4)
+struct CPed : public rage::fwEntity {
```

Comment:
Why prefixing the class names with `C`? That seems odd

Reply:
It's from the game, otherwise I would've omitted it.